src/universal/bin/apalache-mc

```diff
@@ -38,16 +38,24 @@ then
     JVM_ARGS="${JVM_ARGS} -Xmx4096m"
 fi
 
+# Avoid SANY concurrency issues: https://github.com/tlaplus/tlaplus/issues/688
+if [ -z "$TMPDIR" ]; then
+    TMPDIR="$(pwd)/tmp"
+    mkdir -p "$TMPDIR"
+fi
+JAVA_IO_TMPDIR=`mktemp -d -p $TMPDIR -t SANYXXXXXXXXXX`
```

Comment:
Why not skip `-p` (and the above) and rely on it's default behavior:

```
-p temp-dir, --tmpdir=temp-dir
              temp  directory  for  the  file.  This option is a member of the
              tmpdir class of options.

              If this option is not provided, mktemp will use the  environment
              variable  TMPDIR to find a suitable directory.  If these are not
              available, it will fall back to ~/tmp  or  /tmp.   A  <file-pat>
              command line argument containing a directory component will con-
              flict with this option.
```

Reply:
For some reason, it does not work with docker this way. Maybe it's using nix and it has different behavior. But it does not even work with my fix.

Reply:
The above test was failing due to extra protection flags, now using `${TMPDIR:-}`. This works. Let me see how it works without `-p`. It definitely did not work without the additional test for `TMPDIR` above